sqlmesh/core/scheduler.py

```diff
@@ -343,8 +343,8 @@ def start_date(
     Returns:
         Start datetime object.
     """
-    if snapshot.model.start:
-        return to_datetime(snapshot.model.start)
+    if snapshot.start:
+        return to_datetime(snapshot.start)
```

Comment:
you shouldn't need to call to_datetime here because you're setting a datetime on line 367